book/api/websocket.md

```diff
@@ -0,0 +1,1093 @@
+
+Account balance of this validators identity key in lamports. The balance
+is on the highest slot of the currently active fork of the validator.
+
+#### `summary.root_slot`
+| frequency       | type     | example     |
+|-----------------|----------|-------------|
+| *Once* + *Live* | `number` | `275138349` |
+
+The last slot that was rooted. Rooted slots are fully confirmed and
+irreversible, and the rooted slot will never decrease as switching fork
+cannot unroot the slot. The number will not always increase by one, as
+skipped slots do not update the root slot. For example, if the root slot
+goes from `1001` to `1003` it means slot `1002` was skipped.
+
+#### `summary.optimistically_confirmed_slot`
+| frequency       | type     | example     |
+|-----------------|----------|-------------|
+| *Once* + *Live* | `number` | `275138349` |
+
+The highest slot on the current fork that was optimistically confirmed.
+Optimistic confirmation means that over two-thirds of stake have voted
+to confirm the slot, and it is unlikely (although still possible, if
+validators switch vote) to not become rooted.
+
+Although rare, the `optimistically_confirmed_slot` could decrease if a
+validator switches to another fork that does not have this slot.
+
+#### `summary.completed_slot`
+| frequency       | type     | example     |
+|-----------------|----------|-------------|
+| *Once* + *Live* | `number` | `275138349` |
+
+The highest completed slot on the current fork choice of the validator.
+The completed slot may decrease if the validator is switching forks, or
+could stay the same for much more than the slot production time (400
+milliseconds) if leaders are offline and not producing blocks.
+
+#### `summary.estimated_slot`
+| frequency       | type     | example     |
+|-----------------|----------|-------------|
+| *Once* + *Live* | `number` | `275138349` |
+
+The estimated slot is the same as the completd slot, except it still
+progresses forward even if the current leaders are skipping (not
+producing) their slot. For example, if the last completed slot was
+`1001` and it has been 800 milliseconds since that slot, the estimated
+slot is likely to be `1003`.
+
+#### `summary.estimated_slot_duration_nanos`
+| frequency       | type     | example     |
+|-----------------|----------|-------------|
+| *Once* + *Live* | `number` | `450267129` |
+
+The estimated duration of each slot on the network. This is a moving
+average from the prior 750 slots, or around five minutes. Live here
+means the estimate is republished whenever it changes, which is when
+a new slot is confirmed on the currently active fork.
+
+#### `summary.skip_rate`
+| frequency       | type       | example                                 |
+|-----------------|------------|-----------------------------------------|
+| *Once* + *Live* | `SkipRate` | `{"epoch": 522, "skip_rate": 0.456172}` |
+
+The skip rate of an epoch is the ratio of `skipped_slots/total_slots`
```

Comment:
What does it publish if `total_slots==0`?

Reply:
0.  Will clarify.